gradlew

```diff
@@ -28,7 +28,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```

Comment:
This is the default, automatically generated by the Gradle wrapper.

Reply:
That small heap size is fine for Gradle.

Reply:
Yes, this the the default generated by gradle wrapper version update. 
I decided to leave it based on idea that Gradle knows better how to update itself.

Reply:
If you’re optimizing for build cache (and for gradle runtimes in general), I would think raising this might be beneficial. Particularly, for compiling scala code I usually see it recommended that the gradle JVM get more memory.

I would personally suggest `-Xmx1024m -XX:MaxPermSize=256m`, but I also see many articles recommending at least `DEFAULT_JVM_OPTS="-Xmx512m"`.

Your call I guess, but I’d consider being sure that the values you’re using there as the default are still being used currently despite being explicitly zero’d out currently.

Reply:
This is not the heap size used for the build and Gradle Daemon. My understanding is that these are the settings used for the Gradle client. See more info on configuring the heap size here: https://docs.gradle.org/current/userguide/build_environment.html#sec:configuring_jvm_memory

The iceberg project has the heap size already configured in the `gradle.properties` here: https://github.com/apache/iceberg/blob/master/gradle.properties#L19

Reply:
> If you’re optimizing for build cache (and for gradle runtimes in general), I would think raising this might be beneficial. Particularly, for compiling scala code I usually see it recommended that the gradle JVM get more memory.

@runningcode is correct.

There's the client VM (environment variables DEFAULT_JVM_ARGS, GRADLE_OPTS and JAVA_OPTS all affect this), the Gradle daemon (controlled by org.gradle.jvmargs which can be set in several ways) and the _worker processes_ used by compilers (built in default, but adjustable in the build script). 

So to affect Scala compilation, you'd have to tweak the last one (worker process).